Core/Core/Views/GradeCircleView.swift

```diff
@@ -71,7 +71,7 @@ public class GradeCircleView: UIView {
             return
         }
 
-        guard submission.grade != nil else {
+        guard submission.grade != nil || submission.excused == true else {
```

Comment:
Does this view hide itself?  A view is usually responsible for drawing itself.  This may be my own opinion but a superview is usually in charge of adding the subview, displaying where it is located and if it's hidden or not.  This is a knitpick won't hold up the review for it.  This logic seems more appropriate at the superview level.